File: playground/BOJ/boj_2747/boj_2747_v1.js
/*----------------------------------------------
Sub  : [BOJ] 피보나치 수
Link : https://www.acmicpc.net/problem/2747
Level: Bronz 2
Tag  : JS, 수학, 피보나치, 다이나믹
Memo
 - 
----------------------------------------------*/

const fs = require('fs')
const input = fs.readFileSync(0, 'utf-8').trim().split('\n')

const n = parseInt(input);

// fibonacci 함수
function fibonacci(n) {
    if (n === 0) {
        return 0;
    } else if (n === 1) {
        return 1;
    }

    let a = 0, b = 1;
    for (let i = 2; i <= n; i++) {
        let temp = b;
        b = a + b;
        a = temp;
    }
    return b;
}

// n번째 피보나치 수 출력
console.log(fibonacci(n));

File: Online_Judge_Problems/BOJs/boj_2747/boj_2747_v1.py
'''---------------------------------------------
Sub  : [BOJ] 피보나치 수
Link : https://www.acmicpc.net/problem/2747
Level: Bronze 2
Tag  : Python, 수학, 구현, 다이나믹
Memo
 - a, b = b, a+b에서 b = a+b -> 이전에 구했던 a의 값을 사용
---------------------------------------------'''

import sys
input = sys.stdin.readline

def fibonacci(n):
    if n == 0:
        return 0
    elif n == 1:
        return 1
    
    a, b = 0, 1
    for _ in range(2, n + 1):
        a, b = b, a+b
    return b

# 입력 받기
n = int(input().strip())

# n번째 피보나치 수 출력
print(fibonacci(n))
